typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -209,6 +227,13 @@ async function main(address: string, options: { output: string }): Promise<void>
         resolve();
       });
     });
+
+    const waitForWriteQueue = async () => {
+      while (writeMsgQueue.getPendingLength() + writeMsgQueue.getQueueLength() > 0) {
+        await new Promise((resolve) => setTimeout(resolve, 100));
+      }
+    };
+    await waitForWriteQueue();
```

Comment:
I don't think you need the wrapper function here. You can just have the `while` loop.